#!/usr/bin/env python
PACKAGE = "leader_follower_task"

import roslib;
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("desired_offset_x", double_t, 0, "X-axis offset between leader and follower", -6.0, -10.0, 10.0)
gen.add("desired_offset_y", double_t, 0, "Y-axis offset between leader and follower", 0.0, -10.0, 10.0)
gen.add("desired_offset_z", double_t, 0, "Z-axis offset between leader and follower", 5.0, -10.0, 10.0)
gen.add("heading_offset", double_t, 0, "heading offset", 0.0, -3.2, 3.2)
gen.add("filter_data", bool_t, 0, "Use Kalman filter to reduce leader position noise and predict velocity", False)
gen.add("use_trajectory_reference", bool_t, 0, "Use trajectory reference instead of a single point", False)
gen.add("use_velocity_reference", bool_t, 0, "Use velocity reference instead of position/trajectory", False)

exit(gen.generate(PACKAGE, "leader_follower_task", "Follower"))
